### PR TITLE
Add "View Submissions" button on Problem admin

### LINKED ISSUE #488: "View submissions" button on problem admin
Would bring up a search filtered by that problem code &mdash; not precisely ideal, but it should work.

### PR DESCRIPTION
**Pull request by Ninjaclasher**
Fixes #488.

### CODE REVIEW

File: judge/admin/submission.py [hunk 1]
```diff
@@ -124,6 +124,9 @@ def has_change_permission(self, request, obj=None):
             return True
         return obj.problem.is_editor(request.user.profile)
 
+    def lookup_allowed(self, key, value):
+        return super(SubmissionAdmin, self).lookup_allowed(key, value) or key in ['problem__code']
```
Comment (Xyene) [inline judge/admin/submission.py]: Using a tuple is more idiomatic in this case.

Comment (quantum5) [reply judge/admin/submission.py]: Tuples are also faster for this kind of thing, but that's not a significant concern. It is more idiomatic and that's what should matter.

File: templates/admin/judge/problem/change_form.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        django.jQuery(function ($) {
+            $('.submissionslink').appendTo('div#bottombar').show();
+        });
+    </script>
+{% endblock extrahead %}
+
+{% block after_field_sets %}{{ block.super }}
+    {% if original %}
+        <a style="display: none" title="{% trans "View Submissions" %}" href="{% url 'admin:judge_submission_changelist' %}?problem__code={{ original.code }}"
+           class="button submissionslink">
```
Comment (quantum5) [inline templates/admin/judge/problem/change_form.html]: Can we keep to convention and use hyphenated CSS class names?

File: templates/admin/judge/problem/change_form.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    <script>
+        django.jQuery(function ($) {
+            $('.submissionslink').appendTo('div#bottombar').show();
+        });
+    </script>
+{% endblock extrahead %}
+
+{% block after_field_sets %}{{ block.super }}
+    {% if original %}
+        <a style="display: none" title="{% trans "View Submissions" %}" href="{% url 'admin:judge_submission_changelist' %}?problem__code={{ original.code }}"
```
Comment (quantum5) [inline templates/admin/judge/problem/change_form.html]: Line too long, swap `href` with `class` so that `href` gets the whole line to itself.

**Pull request merged**